### ISSUE TITLE
Sub suggestions

### ISSUE THREAD

**Issue by mitm001** [SoftbearStudios/mk48#201]
1. Subs reload way to slow in comparison to ships. At minimum a 10 to 20 % boost in reload capabilities is called for. Ships regain fire power and health at a significantly faster pace due to being on the surface since chests boost health and float.
2. All ships should collect exp based off each hit to the target and the amount of dame it does. I cant count the number of times I beat a high level ship down to almost sunk and someone else just one hits them for all the exp. Not to mention all the times you sink the ship and someone else sucks up all the chests. Especially when attacking a fleet. You sink their team mates and the fleet just gets stronger based off your efforts as they suck up all the chests for exp, health gains and quicker reloads. Theirs no logic in a fleets gaining strength from losing battles but that's what happens. 

Having each hit give exp based off damage done will distribute the exp fairly. This requires striping exp gain from chests which would also keep exp campers off the leader board as they would have to actually attack something to get exp.

**Comment by finnbear** [comment]
I agree that the current situation is a bit unfair, seeing as you could take a battleship from 100% health to 10% health and have another player finish them off and claim the score. However, as health can be regenerated, we can't award points proportional to the target's score prior to their death (to prevent infinite points) unless it was implemented as literally stealing their points. A different solution would be to track 2-3 players who recently did the most damage and split the xp reward between them when the kill is successful.

**Comment by mitm001** [comment]
Yes, you can use a unsorted list (hash table in java) to track all damage done before being sunk. Just continue to add the damage to the top of the list with each hit. 

Then use the exp you allow for that victim to determine the top number of players who get the exp starting with the top of the table and working down. 

That means it would allow only the most recent damage given to count towards the split not necessarily just two or three players. This would also account for any healing fairly.

**Comment by mitm001** [comment]
Its pretty apparent you have coded this game to favor surface ships. The problem is that you hamstring subs rather than taking into account surface ships are very powerful if the player understands them. Not to mention fleets of them moving around slaughtering everything in site. 

Lowering the reload time for all subs by 20% is definitely not to extreme and would serve to level the playing field a little against fleets. Implementing the exp solution above would also help alot. Subs are distance attackers so lose all exp from the floating chest almost every time and heal their attackers friends when they do sink a ship if they are in a fleet (the severely flawed heal system you use). 

Even low level ships are tough to beat if the players learns them. You guys even made planes so strong they rarely miss probably catering to some players who didnt understand that carriers are not meant to be a solo killing machine but rather they are a tool for a fleet. I used to be able to take one out with a level 2 sub if they were solo but not anymore. It takes an Ohio or you have to catch them in a weakened state if they are solo. They should be easy targets if they are being used like that. They were never in the history of navies used as a solo killing machine. They were always part of a fleet, period. If they were off on their own they would get sunk easily as they were not designed for that type of use.

The one thing you rarely see is a fleet of subs running around. Ask yourself why that is. Its because they are meant for solo operations. At best one can assist a fleet but even then you will notice players are not really using them these days for fleets as they get sunk to easily. The level 8 destroyer is used almost exclusively now. This is due to how you have screwed over subs in favor of ships.

**Comment by finnbear** [comment]
> Its pretty apparent you have coded this game to favor surface ships.

Our intention is to emphasize realistic tradeoffs, while keeping the game balanced.

The natural consequence is that ships have certain benefits (e.g. usually more weapons/reloads, ability to deploy aircraft, more health, get surface XP easily) and submarines have certain other benefits (e.g. stealthy, immunity from most weapons when submerged, sonar decoys, generally better torpedoes).

> The level 8 destroyer is used almost exclusively now

For several months, I've been planning to add a log counting the relative frequency of each ship type among players, and maybe even their score. This could help objectively determine if anything is overpowered (or underpowered).

> This is due to how you have screwed over subs in favor of ships.

That isn't our intention. However, it is true that we have taken steps to avoid situations inwhich ships of all shapes and sizes die  to a sub that became visible moments earlier, because that is extremely toxic and likely to cause rage-quitting.

> Lowering the reload time for all subs by 20% is definitely not to extreme

It is true that certain submarines have automatic loading systems for their torpedo tubes, which implies they should be able to reload tubes in parallel. Right now, it's sequential.

**Comment by mitm001** [comment]
> reload tubes in parallel.

Thats interesting. That is something that may be an improvement. Ships reload crazy fast compared to subs. 

This brings up another point I forgot. Subs need more anti-aircraft. Yasen needs 2 more and the level 6 sub needs 2 more or maybe cut the reload time. The planes and drones reload significantly faster and you can only get about 50% if your lucky. This is assuming the enemy ship is so far away you cant see them on screen. If you can see them its probably to late to defend against them.

**Comment by mitm001** [comment]
Ill put this gently, subs need a little love. 

Its getting ridiculous when you have a carrier that is anti-sub. Everything is anti-sub and super stealth vs subs. I have said it before but its worth mentioning again. Subs rule the seas, no ifs and or buts. Your level 2 sub could easily sink a carrier in real life. Theirs not a ship afloat after WW2 that can win against a sub, 0, nada, zilch, none, they do not exist, period. 

They really need to be beefed up.

You should reconsider allowing fleets to interact with non fleet players also. This is a major issue now that you can have 6 players roaming and slaughtering everything in sight. I assure you that you cannot monetize any game that allows this kind of interaction, 6 vs 1.

The exp thing is really a big drawback considering its so simple to actually implement a fair system using tables/lists. Subs cannot afford to get in close combat with ships if they want to survive and that's the only way they can heal and gain all the exp they should get from a sinking. Its makes no sense a sub can sink an enemy in a fleet and actually heal and rearm the remaining members as well as be out of ammo for doing it. 

Ship upgrading is still a mess even though its not as bad as it was. Any sub that is not at full health when it upgrades gets severely penalized, usually to all zeros except the decoys. You cannot upgrade a sub in battle but ships can do so and not have a problem as they reload to a deadly state faster.

Speaking of upgrading, healing a player from upgrading is a flawed system in and of itself. In a fair situation, if someone upgrades in battle to save their neck and gets healed then the opponents should all have their weapons recharged. I lose a lot of sinkings to someone upgrading with needing only one hit to sink, they upgrade and all I can do is watch them go to half health as weapons reload so slow and are so weak you don't stand a chance of getting them afterwards. Ship upgrading would be best if it did not heal at all.

An Akula can dump 8 belly shots into an Iowa and not sink it. Do you think that's even possible? The answer is no.
